### PR TITLE
:wrench: Use yamllint to check lint issues on GH workflows

### PR DESCRIPTION
**Pull request by mauromorales**
This PR adds an Earhtly target to lint the GH workflow yaml files plus it fixes the existing issues

### CODE REVIEW

**Review by Itxaka — APPROVED** [review]
niiiice

**Comment by mudler** [comment]
no more far west! :)

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -19,4 +19,7 @@ jobs:
           go-version: '^1.18'
       - name: Run Lint checks
         run: |
-          ./earthly.sh +lint
\ No newline at end of file
+          ./earthly.sh +lint
+      - name: Yaml Lint checks
```
Comment (sdwilsh) [inline .github/workflows/lint.yml:23]: Why not add this to the overall lint target instead?

Comment (sdwilsh) [reply .github/workflows/lint.yml:23]: I see this was addressed in 44cfacd819f08a905fb2596065bf988a2b8f8226 :)

**Pull request merged**